File: Java/_003_Longest_Substring_Without_Repeating_Characters/Solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] map = new int[256];
        Arrays.fill(map, -1);
        int maximum = 0;
        int start = -1;
        for (int i = 0; i < s.length(); i++)
        {
            if (map[s.charAt(i)] > start)
                start = map[s.charAt(i)];
            map[s.charAt(i)] = i;
            maximum = Math.max(maximum, i - start);
        }
        return maximum;
    }
}

File: Python/_003_Longest_Substring_Without_Repeating_Characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        mapping = [-1] * 256
        maximum = 0
        start = -1
        for i in range(len(s)):
            if mapping[ord(s[i])] > start:
                start = mapping[ord(s[i])]
            mapping[ord(s[i])] = i
            maximum = max(maximum, i - start)
        return maximum
